File: 0443-string-compression/0443-string-compression.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function(chars) {
  const n = chars.length;
  let [start, i] = [0, 0];
  
  while (i < n) {
    const char = chars[i];
    let count = 0;
    while (i < n && chars[i] === char) {
      i++;
      count++;
    }
    
    chars[start++] = char;
    if (count > 1) 
      for (let ch of `${count}`) chars[start++] = ch;
  }
  
  return start;
};

File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        n = len(chars)
        start = i = 0
        while i < n:
            char, count = chars[i], 0
            while i < n and chars[i] == char:
                i += 1
                count += 1
            chars[start] = char
            start += 1
            if count > 1:
                for ch in str(count):
                    chars[start] = ch
                    start += 1
        return start